### PR TITLE
Nodejsversions

### PR DESCRIPTION
**Pull request by mhdawson**

### CODE REVIEW

File: docs/functional-components/nodejs-versions-images.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+| Binaries with FIPs support        | OS Packages from OS vendor                                    |
+| Containers                        | Official Docker images excluding Alpine                       |
+| Commercially Supported Containers | Images from distros (for example rhel or ubi images)          |
+| Containers with FIPs support      | Images from distros (for example rhel or ubi images)          |
+
+We recommend that you use only [LTS](https://github.com/nodejs/release#release-phases)
+versions for production use, and in addition test on the latest `Current` in order
+to prepare for future upgrades. A new LTS version is available every October from the
+project and past LTS versions go EOL in April, 30 months after they were released.
+You should
```
Comment (lholmquist) [inline docs/functional-components/nodejs-versions-images.md:20]: looks like a Rogue sentence :)

File: docs/functional-components/nodejs-versions-images.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+Node.js version provided by an OS package manager. Instructions
+on this [page](https://nodejs.org/en/download/package-manager/) may
+help in configuring so that you get an up to date version.
+
+For Red Hat and IBM deployments that need commercial support
+we recommend the binaries which come with RHEL.
+
+### Binaries with FIPs support  
+
+The community is an awkward period where the only suppored versions
```
Comment (lholmquist) [inline docs/functional-components/nodejs-versions-images.md:69]:
```suggestion
The community is at an awkward period where the only supported versions
```

File: docs/functional-components/nodejs-versions-images.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+include the Node.js binaries.
+
+For example.  if you want to take advantage of the support for
+Node.js that comes with RHEL or RedHat Runtimes you will need to use
+the rhel or ubi images available from the
+[Red Hat container catalog](https://catalog.redhat.com/)
+
+### Containers with FIPs support  
+
+The community is an awkward period where the only suppored versions
```
Comment (lholmquist) [inline docs/functional-components/nodejs-versions-images.md:126]:
```suggestion
The community is at an awkward period where the only supported versions
```

File: docs/functional-components/nodejs-versions-images.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+### Raw binaries
+
+The Node.js projects provides binaries for the 
+supported platforms on the [download](https://nodejs.org/en/download/) page.
+
+These are the most up to date packages available and new releases show
+up on this page first.
+
+### Windws and Mac installers
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:30]:
```suggestion
### Windows and Mac installers
```

File: docs/functional-components/nodejs-versions-images.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+These are the most up to date installers available and new releases show
+up on this page first.
+
+### Node.js version managers
+
+Node.js version managers provide an easier way to get Node.js
+versions from the Node.js download site and switch between them.
+The the teams organizations have the most experience with
+[nvm](https://github.com/nvm-sh/nvm)
+with the caveat that it does not support windows and that
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:44]:
```suggestion
with the caveat that it does not support Windows and that
```

File: docs/functional-components/nodejs-versions-images.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+### Commercially Supported Binaries
+
+Most often the easiest way to get support is from an operating system
+vendor. This generally requires using the binaries which are 
+installed using the native package managers for that OS.
+
+**Note:** It is quite common for the version of Node.js which
+is installed in this way to be tied to the OS version.
+for example if you simply install `apt-get install nodejs` on
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:56]:
```suggestion
For example if you simply install `apt-get install nodejs` on
```

File: docs/functional-components/nodejs-versions-images.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  * debian
+  * debian-minimal
+  * alpine
+
+There are two image flavors based on debian so that you can achieve smaller container
+sizes using [multi-stage docker builds](https://docs.docker.com/develop/develop-images/multistage-build/).
+The larger images has all of the tools need to build the application while
+the minimal image has only the base components needed to run Node.js. In both cases
+these images bundle in the binaries which are available on the Node.js download
+site and support multiple ptatforms (x64, PPC and s390)
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:101]:
```suggestion
site and support multiple ptatforms (x64, PPC and s390).
```

File: docs/functional-components/nodejs-versions-images.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+Images are also provided for Alpine which can allow you to achieve even smaller
+images sizes. **However**, if you check the
+[BUILDING.md](https://github.com/nodejs/node/blob/master/BUILDING.md)
+file for the Node.js project you'll see that support for Alpine is `experimental`.
+This also means that there is no binary which can be bundled and instead
+the creation of the Alpine containers includes building the binaries
+themselves. For there reasons we don't recommend using the Alpine images
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:109]:
```suggestion
themselves. For these reasons we don't recommend using the Alpine images
```

File: docs/functional-components/nodejs-versions-images.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+unless the smaller size is an absolute requirement.
+
+### Commercially Supported Containers
+
+Most often the easiest way to get support is from an operating system
+vendor. This generally requires using a host running an operating
+system from the vendor as well as containers from that vendor which
+include the Node.js binaries.
+
+For example.  if you want to take advantage of the support for
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:119]:
```suggestion
For example, if you want to take advantage of the support for
```

File: docs/functional-components/nodejs-versions-images.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+Node.js that comes with RHEL or RedHat Runtimes you will need to use
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:120]:
```suggestion
Node.js that comes with RHEL or Red Hat Runtimes you will need to use
```

File: docs/functional-components/nodejs-versions-images.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+Always cache the binaries, containers or OS packages tha you use in
+production and in build/test pipelines. For example, do not depend
+on the Node.js download site being available 24/7.
+
+Subscribe to the nodejs-sec mailing list. This low volumne
+mailing list is used to provide advance notice of security releases
+and will give you the earliest warning that you may need
+to update your Node.js version.
+
+For new applications alwasys start with the latest LTS version.
```
Comment (richardlau) [inline docs/functional-components/nodejs-versions-images.md:154]:
```suggestion
For new applications always start with the latest LTS version.
```

File: docs/functional-components/nodejs-versions-images.md (hunk 1 above)
Comment (mhdawson) [reply docs/functional-components/nodejs-versions-images.md]: Fixed

**Comment by roastlechon** [comment]
should we bring up tools like `nvm` or `fnm`

**Comment by mhdawson** [comment]
@roastlechon I tried to cover that in:

**Comment by mhdawson** [comment]
ping @roastlechon I think your question is the last one I need to address before landing. Cloud you take a quick look to see if you agree that I've covered what you were asking about.

**Comment by roastlechon** [comment]
> ping @roastlechon I think your question is the last one I need to address before landing. Cloud you take a quick look to see if you agree that I've covered what you were asking about.

Reviewing now

**Review by roastlechon — APPROVED** [review]

**Comment by mhdawson** [comment]
We discussed in the last Ref Arch team meeting and feedback was that it looked good. Landing

**Pull request merged**